wiring/inc/spark_wiring_system.h

```diff
@@ -1006,6 +1058,25 @@ class SystemClass {
 
 extern SystemClass System;
 
+inline SystemHardwareInfo::SystemHardwareInfo()
+        : info_{},
+          error_(SYSTEM_ERROR_INVALID_STATE) {
+    memset(&info_, 0xff, sizeof(info_));
+    info_.size = sizeof(info_);
+    info_.version = 0;
+    memset(info_.ncp, 0xff, sizeof(info_.ncp));
+    // FIXME: this workaround should better be part of dynalib interface, but because of dependencies
+    // between hal/system/wiring/etc for now it's easier to deal with it here, and we don't have that many cases
+    // like this one.
+    // NOTE: hal_get_device_hw_info() is not available in 5.x releases <= 5.0.0
+    if (System.versionNumber() >= SYSTEM_VERSION_v500ALPHA1 && System.versionNumber() <= SYSTEM_VERSION_v500) {
```

Comment:
Instead of backporting the `SYSTEM_VERSION_5*` defines, could you use `SYSTEM_VERSION_ALPHA(5, 0, 0, 1)` and `SYSTEM_VERSION_DEFAULT(5, 0, 0)` ?  This would keep 4.x bump commits very familiar, without the trailing `SYSTEM_VERSION_5*` defines.

Reply:
I don't have a strong preference on this, but it makes it easier on the eyes we could also just move the defines below the `SYSTEM_VERSION` that gets updated.

Reply:
I'm going to merge as-is and just shuffle the defines when doing a bump commit.